### PR TITLE
java exceptions with optional linking

### PR DESCRIPTION
**Pull request by gak**

### CODE REVIEW

File: src/services/log.ts [hunk 1]
```diff
@@ -432,11 +457,70 @@ export function makeCopyBtn(v: any, copyBag: Array<any>): string {
...
+
+interface JavaExceptionConfig {
+  matches: JavaExceptionMatch[]
+}
+
+function fqcnMatches(fqcn: string, config: JavaExceptionConfig): JavaExceptionMatch | undefined {
+  return config.matches.find(m => fqcn.match(new RegExp(m.fqcnMatch)))
+}
+
+function javaException(config: JavaExceptionConfig): ExpandedTransform {
```
Comment (mightyguava) [inline src/services/log.ts:475]: I'm getting a bit lost, where does JavaExceptionConfig get passed in?

File: src/services/log.ts [hunk 2]
```diff
@@ -432,11 +457,70 @@ export function makeCopyBtn(v: any, copyBag: Array<any>): string {
...
+  fqcnMatch: string
+  // The link generated for a matched fqcn
+  href: string
+}
+
+interface JavaExceptionConfig {
+  matches: JavaExceptionMatch[]
+}
+
+function fqcnMatches(fqcn: string, config: JavaExceptionConfig): JavaExceptionMatch | undefined {
```
Comment (mightyguava) [inline src/services/log.ts:471]: inline this function please, easier to read that way, or maybe some comments if this adds some outsized value?

File: src/services/log.ts [hunk 3]
```diff
@@ -432,11 +457,70 @@ export function makeCopyBtn(v: any, copyBag: Array<any>): string {
...
+        let right = `${fqcn}(${file}${lineno || ""})`
+        // TODO: This will have to be customisable when we open source
+
+        const exceptionMatch = fqcnMatches(fqcn, config)
+
+        if (exceptionMatch) {
+          const parts = fqcn.split('.')
+          const firstCapital = parts.findIndex((i: string) => i[0].toLowerCase() !== i[0])
+          const path = parts.slice(0, firstCapital).join('/')
+          const href = exceptionMatch.href
```
Comment (mightyguava) [inline src/services/log.ts:499]: so `href` is a "template string" that you pass in via a config somewhere? Tests and/or documentation would be really useful here.

File: src/services/log.ts (hunk 2 above)
Comment (gak) [reply src/services/log.ts:471]: https://github.com/cashapp/logquacious/pull/23

File: src/services/log.ts (hunk 3 above)
Comment (gak) [reply src/services/log.ts:499]: I have a ticket for updating the README about this. Similar with the other comment: https://github.com/cashapp/logquacious/pull/10/files/022670f93aa7feef2d7a60f2a7cb9a7879eb4b92#r348552561

**Pull request merged**